### PR TITLE
Backport new kernel patches

### PR DESCRIPTION
**Pull request by keboliu**
0024-mlxsw-core-Fix-memory-leak-on-module-removal.patch https://github.com/torvalds/linux/commit/adc80b6cfedff6dad8b93d46a5ea2775fd5af9ec  @v5.10-rc2
0025-platform-x86-mlx-platform-Remove-PSU-EEPROM-configur.patch  https://github.com/torvalds/linux/commit/2bf5046bdb649908df8bcc0a012c56eee931a9af @v5.11-rc1
0026-platform-x86-mlx-platform-Remove-PSU-EEPROM-configur.patch  https://github.com/torvalds/linux/commit/912b341585e302ee44fc5a2733f7bcf505e2c86f @v5.11-rc1

regression tests have been performed against these patches on the Mellanox platform, no issue found.

Signed-off-by: Kebo Liu <kebol@nvidia.com>

### CODE REVIEW

File: patch/0026-platform-x86-mlx-platform-Remove-PSU-EEPROM-configur.patch [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+address space two bytes. Thus, all these devices have been handled by
+EEPROM driver "24c02".
+There is a new requirement is to support power unit replacement by "off
+the shelf" device, matching electrical required parameters. Such device
+could be equpped with different EEPROM type, which could be one byte
+address space adressing or even could be not equipped with EEPROM.
+In such case "24c02" will not work.
+
+Fixes: ef08e14a3 ("platform/x86: mlx-platform: Add support for new msn274x system type")
+Signed-ff-by: Oleksandr Shamray <oleksandrs@nvidia.com>
```
Comment (paulmenzel) [inline patch/0026-platform-x86-mlx-platform-Remove-PSU-EEPROM-configur.patch:19]: Signed-off-by line typo.

File: patch/0024-mlxsw-core-Fix-memory-leak-on-module-removal.patch [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    [<000000000e2bc177>] __driver_attach+0x167/0x2f0
+    [<000000007d44cd6e>] bus_for_each_dev+0x148/0x1f0
+    [<000000003cd5a91e>] driver_attach+0x45/0x60
+    [<000000000041ce51>] bus_add_driver+0x3b8/0x720
+    [<00000000f5215476>] driver_register+0x230/0x4e0
+    [<00000000d79356f5>] __pci_register_driver+0x190/0x200
+
+Fixes: a22712a96291 ("mlxsw: core: Fix devlink unregister flow")
+Signed-off-by: Ido Schimmel <idosch@nvidia.com>
+Reported-by: Vadim Pasternak <vadimp@nvidia.com>
```
Comment (paulmenzel) [inline patch/0024-mlxsw-core-Fix-memory-leak-on-module-removal.patch:33]: Please use exactly the commit from Linux master branch.

https://github.com/torvalds/linux/commit/adc80b6cfedff6dad8b93d46a5ea2775fd5af9ec looks different. If the commit hash differs, please put the Linux master commit hash in the beginning [as done for the stable series](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-5.10.y&id=55d900415b81680bcd5f93be51f26ebfc51ac6fb).

**Review by paulmenzel — COMMENTED** [review]
Thank you for the backport. Additionally to the comments, please mention in the merge/pull request description, how you tested this.

**Comment by paulmenzel** [comment]
Also, at least the last patch is in Linux master since v5.11-rc1. Have you sent them for inclusion into the stable series, at least 5.10, so we can drop the patches, once updating to the latest 5.10.y release?

File: patch/0026-platform-x86-mlx-platform-Remove-PSU-EEPROM-configur.patch (hunk 1 above)
Comment (keboliu) [reply patch/0026-platform-x86-mlx-platform-Remove-PSU-EEPROM-configur.patch]: patch has been re-generated with the upstream commit.

File: patch/0024-mlxsw-core-Fix-memory-leak-on-module-removal.patch (hunk 2 above)
Comment (keboliu) [reply patch/0024-mlxsw-core-Fix-memory-leak-on-module-removal.patch:34]: patch has been re-generated with the upstream commit.

**Comment by keboliu** [comment]
> Also, at least the last patch is in Linux master since v5.11-rc1. Have you sent them for inclusion into the stable series, at least 5.10, so we can drop the patches, once updating to the latest 5.10.y release?

I can talk to the author of these patches to see the what we can do.

File: patch/0024-mlxsw-core-Fix-memory-leak-on-module-removal.patch (hunk 2 above)
Comment (keboliu) [reply patch/0024-mlxsw-core-Fix-memory-leak-on-module-removal.patch:34]: patch has been updated.

**Comment by keboliu** [comment]
@paulmenzel could you please review again?

**Comment by liat-grozovik** [comment]
@ paulmenzel can you please mark your comments as resolved and approve? we would like to move forward with the merge and have it on both 202012 and master.

**Comment by liat-grozovik** [comment]
@lguohan  can you please review as well?

**Comment by liat-grozovik** [comment]
@paulmenzel and @lguohan  kindly help to review and merge?
this one is holding us for having bugs fixes in master/202012. please also add the label for 202012 merge if there is already new branch for it.

**Review by lguohan — APPROVED** [review]

**Comment by paulmenzel** [comment]
Sorry for not responding earlier. I thought, you can mark the comments as resolved once you implemented the change. @lguohan, thank you for taking care of this.

**Pull request merged**